### PR TITLE
staticpodstate: improve condition and event messages

### PR DESCRIPTION
**Pull request by mfojtik**
We repeat the node name in pod name, which makes the events very long for no good reason.
This improves the messages and also make the controller to resync periodically, in case the pod informer is stuck.

This also explains the reason for ` container kube-apiserver is not ready`.

### CODE REVIEW

File: pkg/operator/staticpod/controller/staticpodstate/staticpodstate_controller.go [hunk 1]
```diff
@@ -19,10 +21,6 @@ import (
 	"github.com/openshift/library-go/pkg/operator/v1helpers"
 )
 
-var (
-	staticPodStateControllerWorkQueueKey = "key"
```
Comment (sttts) [inline pkg/operator/staticpod/controller/staticpodstate/staticpodstate_controller.go:23]: unused?

File: pkg/operator/staticpod/controller/staticpodstate/staticpodstate_controller.go [hunk 2]
```diff
@@ -88,15 +96,18 @@ func (c *StaticPodStateController) sync(ctx context.Context, syncCtx factory.Syn
 				// When container is not ready, we can't determine whether the operator is failing or not and every container will become not
 				// ready when created, so do not blip the failing state for it.
 				// We will still reflect the container not ready state in error conditions, but we don't set the operator as failed.
-				errs = append(errs, fmt.Errorf("nodes/%s pods/%s container=%q is not ready", node.NodeName, pod.Name, containerStatus.Name))
+				errs = append(errs, fmt.Errorf("pod/%s container %q is not ready: %s", pod.Name, containerStatus.Name, describeWaitingContainerState(containerStatus.State.Waiting)))
```
Comment (sttts) [inline pkg/operator/staticpod/controller/staticpodstate/staticpodstate_controller.go:99]: pod name includes node name?

File: pkg/operator/staticpod/controller/staticpodstate/staticpodstate_controller.go (hunk 1 above)
Comment (mfojtik) [reply pkg/operator/staticpod/controller/staticpodstate/staticpodstate_controller.go:23]: yes

File: pkg/operator/staticpod/controller/staticpodstate/staticpodstate_controller.go (hunk 2 above)
Comment (mfojtik) [reply pkg/operator/staticpod/controller/staticpodstate/staticpodstate_controller.go:99]: yes

**Pull request merged**